components/movementsensor/gpsnmea/serial.go

```diff
@@ -146,12 +148,33 @@ func (g *SerialNMEAMovementSensor) GetCorrectionInfo() (string, uint) {
 
 // Position position, altitide.
 func (g *SerialNMEAMovementSensor) Position(ctx context.Context, extra map[string]interface{}) (*geo.Point, float64, error) {
+	lastPosition := g.lastposition.GetLastPosition()
+
 	g.mu.RLock()
 	defer g.mu.RUnlock()
-	if g.data.location == nil {
-		return geo.NewPoint(0, 0), 0, errNilLocation
+
+	currentPosition := g.data.location
+
+	if currentPosition == nil {
+		return lastPosition, 0, errNilLocation
```

Comment:
[Q] Would a currentPosition of (NaN, NaN) be considered nil?

Reply:
apparently not. I should probably check for NaN